app/components/channel_modal.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PropTypes, PureComponent} from 'react';
+import {Animated, Dimensions} from 'react-native';
+
+const {View: AnimatedView} = Animated;
+const {width: deviceWidth, height: deviceHeight} = Dimensions.get('window');
+
+export default class ChannelModal extends PureComponent {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        duration: PropTypes.number.isRequired,
```

Comment:
My guess is that we have some `defaultProps` probably we don't need to mark them as `Required`? @hmhealey @thomchop @jarredwitt what do you think?